### PR TITLE
Added Torch ENet support

### PR DESCRIPTION
**Pull request by arrybn**
### This pullrequest changes

Added loading Torch ENet model: changed Torch models reader and added all necessary layers

### CODE REVIEW

File: modules/dnn/samples/torch_enet.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
```
Comment (alalek) [inline modules/dnn/samples/torch_enet.cpp]: Usually we don't put license header into sample files.

File: modules/dnn/samples/torch_enet.cpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+    if (inputImgSize != img.size())
+        resize(img, img, inputImgSize);       //Resize image to input size
+
+    if(img.channels() == 3)
+        cv::cvtColor(img, img, cv::COLOR_BGR2RGB);
+
+    img.convertTo(img, CV_32F);
+
+    img /= 255;
```
Comment (alalek) [inline modules/dnn/samples/torch_enet.cpp]: Merge into single line:
```
img.convertTo(img, CV_32F, 1/255.0);
```

File: modules/dnn/src/layers/batch_norm_layer.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// Copyright (C) 2016, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+/*
+Declaration of Batch Normalization layer.
+*/
+
+#ifndef __OPENCV_DNN_LAYERS_BATCH_NORM_LAYER_HPP__
+#define __OPENCV_DNN_LAYERS_BATCH_NORM_LAYER_HPP__
+#include "../precomp.hpp"
```
Comment (alalek) [inline modules/dnn/src/layers/batch_norm_layer.hpp]: `precomp.hpp` should be included from `.cpp` files only.

File: modules/dnn/samples/torch_enet.cpp (hunk 1 above)
Comment (arrybn) [inline modules/dnn/samples/torch_enet.cpp]: Removed, thanks

File: modules/dnn/samples/torch_enet.cpp (hunk 2 above)
Comment (arrybn) [inline modules/dnn/samples/torch_enet.cpp]: fixed, thanks

File: modules/dnn/src/layers/batch_norm_layer.hpp (hunk 3 above)
Comment (arrybn) [inline modules/dnn/src/layers/batch_norm_layer.hpp]: Removed, thanks

**Comment by vpisarev** [comment]
cool! let's put it in! :+1:

**Pull request merged**